### ISSUE TITLE
Variables needed for gradient computation should be modified by an inplace operation.

### ISSUE THREAD

**Issue by aleversn** [microsoft/ProphetNet#38]
In the README.md, it suggests using the torch of version `1.3.0`, but there seems no that version in the previous version of PyTorch, [link](https://pytorch.org/get-started/previous-versions/).

So, I use the latest version (`1.7.1`) of the torch, and when I start training, I got this Runtime Error.
![1](https://user-images.githubusercontent.com/30100324/109978780-30c8b580-7d39-11eb-8868-be9444d9f838.jpg)
And then I found that the error was caused in the `prophetnet/ngram_multihead_attention.py` line 255.
```python
q = q * self.scaling
```
It looks like this operation is not allowed anymore, then I fixed the problem by the following:
```python
q_ = q * self.scaling

if self.bias_k is not None:
    assert self.bias_v is not None
    k = torch.cat([k, self.bias_k.repeat(1, bsz, 1)])
    v = torch.cat([v, self.bias_v.repeat(1, bsz, 1)])
    q = q_.contiguous().view(tgt_len, bsz * self.num_heads, self.head_dim).transpose(0, 1)
```
